cmd/collector/app/server/http.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/handler"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	clientcfgHandler "github.com/jaegertracing/jaeger/pkg/clientcfg/clientcfghttp"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+)
+
+// HTTPServerParams to construct a new Jaeger Collector HTTP Server
+type HTTPServerParams struct {
+	Port            int
+	Handler         handler.JaegerBatchesHandler
+	RecoveryHandler func(http.Handler) http.Handler
```

Comment:
why does this need to be exposed?

Reply:
Not sure I get the question. The `HTTPServerParams` instance is created by the `Collector#Start` in `cmd/collector/app`...

Reply:
the recovery handler is a mechanical thing that can be added silently, it's weird to have it exposed via API like this.